File: C# ad y ad/2_ Orientado a objetos/S05.1_ La observación/Solucion_la_observacion/La_observacion/Src/Persona/Persona.cs
using System;


namespace La_observacion
{
	class Persona
	{
		string nombre;


		public Persona(string nombre)
		{
			this.nombre = nombre;
		}


		public void Observar(Arbol arbol)
		{
			Console.WriteLine(nombre + " ve un arbol con una altura de " + arbol.GetAltura() + " m. Cantidad de hojas: " + arbol.GetCantidadDeHojas() + ".");
		}
		public void Observar(Piedra piedra)
		{
			Console.WriteLine(nombre + " observa una piedra. Su color: " + piedra.GetColor() + ". Su tamanio: " + piedra.GetTamanio() + ".");
		}
		public void Observar(Nube nube)
		{
			Console.WriteLine(nombre + " mira a una nube en el cielo. Su color: " + nube.GetColor() + ".");
		}
	}
}

File: C++ ad y ad/3_ Orientado a objetos/S05.1_ La observacion/Solucion_la_observacion/La_observacion/src/persona/persona.cpp
#include "persona.h"

#include <iostream>

using namespace std;


Persona::Persona(string nombre)
{
	this->nombre = nombre;
}
Persona::~Persona()
{
	cout << "La persona con el nombre " << nombre << " fue liberada de la memoria." << endl << endl;
}


void Persona::observar(Arbol* arbol)
{
	cout << nombre << " ve un arbol con una altura de " << arbol->getAltura() << " m. Cantidad de hojas: " << arbol->getCantidadDeHojas() << "." << endl;
}
void Persona::observar(Piedra* piedra)
{
	cout << nombre << " observa una piedra. Su color: " << piedra->getColor() << ". Su tamanio: " << piedra->getTamanio() << "." << endl;
}
void Persona::observar(Nube* nube)
{
	cout << nombre << " mira a una nube en el cielo. Su color: " << nube->getColor() << "." << endl;
}
